Update for adding Codepage 850 conversion.

LaZar00 opened the pull request:
This is a PR for adding Codepage 850 to SCI Companion v3 (in case you are interested in update it).

Kawa-oneechan commented on SCICompanionLib/SCICompanionLib.rc:
```diff
@@ -3112,6 +3112,7 @@ END
 IDD_NEWGAMEDIALOG DLGINIT
 BEGIN
     IDC_COMBOLANGUAGE, 0x403, 8, 0, "DOS 437\0",
```
At this point, these two lines and the other pair later on should be renamed "DOS US" and "DOS Europe" or something like that.

Kawa-oneechan commented on SCICompanionLib/Src/Dialogs/GamePropertiesDialog.cpp:
```diff
@@ -56,13 +56,18 @@ void CGamePropertiesDialog::DoDataExchange(CDataExchange* pDX)
 
+	// Change for accepting Codepage 850
+	// m_wndComboLanguage.SetCurSel(appState->GetResourceMap().Helper().GetCodepage() == 1252 ? 2 : 1);
+	m_wndComboLanguage.SetCurSel(appState->GetResourceMap().Helper().GetCodepage() == 1252 ? 2 :
```
Would highly suggest using a temporary variable to hold the value returned by `GetCodePage` and using that instead of calling the function twice.

Kawa-oneechan commented on SCICompanionLib/Src/Dialogs/GamePropertiesDialog.cpp:
```diff
@@ -205,7 +210,9 @@ void CGamePropertiesDialog::OnOK()
 	*/
-	appState->GetResourceMap().Helper().SetCodepage(m_wndComboLanguage.GetCurSel() == 1 ? 1252 : 437);
+	//appState->GetResourceMap().Helper().SetCodepage(m_wndComboLanguage.GetCurSel() == 1 ? 1252 : 437);
+	appState->GetResourceMap().Helper().SetCodepage(m_wndComboLanguage.GetCurSel() == 2 ? 1252 : 
```
Same goes for `GetCurSel`.

Kawa-oneechan replied:
Same goes for `GetCurSel`.

Kawa-oneechan commented on SCICompanionLib/Src/Dialogs/NewGameDialog.cpp:
```diff
@@ -228,7 +228,9 @@ void NewGameDialog::OnBnClickedOk()
 			helper.SetIniString(GameSection, LanguageKey, (lang == LangSyntaxSCI) ? LanguageValueSCI : LanguageValueStudio);
 
-			helper.SetIniString(GameSection, CodepageKey, (m_wndComboLanguage.GetCurSel() == 1) ? "1252" : "437");
+			// helper.SetIniString(GameSection, CodepageKey, (m_wndComboLanguage.GetCurSel() == 1) ? "1252" : "437");
```
As before, same suggestion.

Kawa-oneechan commented on SCICompanionLib/Src/Resources/Font.cpp:
```diff
@@ -194,20 +195,29 @@ EGACOLOR egaFontPalette[] =
+		if (nCel < 0x80)
+			StringCchPrintf(pszLabel, cch, TEXT("%c (%X)"), nCel, nCel);
+		else
+			StringCchPrintf(pszLabel, cch, TEXT("%c  %c  (%X)"), nCel, Dos2WinChar(nCel), nCel);
```
I like this one.

Kawa-oneechan commented on SCICompanionLib/Src/Dialogs/NewGameDialog.cpp:
```diff
@@ -228,7 +228,9 @@ void NewGameDialog::OnBnClickedOk()
 
-			helper.SetIniString(GameSection, CodepageKey, (m_wndComboLanguage.GetCurSel() == 1) ? "1252" : "437");
+			// helper.SetIniString(GameSection, CodepageKey, (m_wndComboLanguage.GetCurSel() == 1) ? "1252" : "437");
+			helper.SetIniString(GameSection, CodepageKey, (m_wndComboLanguage.GetCurSel() == 2) ? "1252" : 
```
As before, same suggestion.

Kawa-oneechan commented on SCICompanionLib/Src/Dialogs/GamePropertiesDialog.cpp:
```diff
@@ -56,13 +56,18 @@ void CGamePropertiesDialog::DoDataExchange(CDataExchange* pDX)
 
+	// Change for accepting Codepage 850
+	// m_wndComboLanguage.SetCurSel(appState->GetResourceMap().Helper().GetCodepage() == 1252 ? 2 : 1);
+	m_wndComboLanguage.SetCurSel(appState->GetResourceMap().Helper().GetCodepage() == 1252 ? 2 :
```
Would _highly_ suggest using a temporary variable to hold the value returned by `GetCodePage` and using that instead of calling the function twice.

LaZar00 replied:
I thought about this. I will try to modify that.

Kawa-oneechan commented on SCICompanionLib/SCICompanionLib.rc:
```diff
@@ -3112,6 +3112,7 @@ END
 IDD_NEWGAMEDIALOG DLGINIT
 BEGIN
     IDC_COMBOLANGUAGE, 0x403, 8, 0, "DOS 437\0",
+    IDC_COMBOLANGUAGE, 0x403, 8, 0, "DOS 850\0",
```
At this point, these two lines and the other pair later on should be renamed "DOS US" and "DOS Europe" or something like that. The only trick is that the 8 in front has to be adjusted to match.

LaZar00 replied:
About this, I had thought about the original codes like OEM-437/OEM-850 in the old systems, or something like CP437/CP850. If you want something that is more understandable like your choices, no problem. We can use that. But DOS Europe may be a little confusing. Greek is Europe and it is not CP850 I think. Or other countries.

Kawa-oneechan replied:
The problem is that technically both of them support European languages. There's a significant overlap between them, but 850 adds more characters in place of certain graphics like ╖ and ▌. Greek would be [CP-737](https://en.wikipedia.org/wiki/Code_page_737), and has no overlap at all. I mean 437 has some Greek in it but those characters are in different spots for 737.

Perhaps "DOS US-Intl" and "DOS Latin-1"?

LaZar00 replied:
I always (I'm near 50, so...) known them as CP437, CP850, or the number itself. The DOS Latin, or descriptions like this were adapted later I think, when appear modern systems like WinXP. I think the best is use a description (it does not matter which). We can use yours proposed or the one I left, but I think that leave the number of the codepage is necessary. To say DOS Latin-1 for example is not so obvious (at least for me). For US, as I understand, there is only one codepage, 437. So that's easy.

I can leave "DOS US-Intl (437)" and "DOS Latin-1 (850)" to leave them more clear. In any case I will leave the width of the combolist as it is, to give a bit more of space in case we need it.

Kawa-oneechan reviewed (changes requested):
This looks good, but I have some suggestions before I merge.

LaZar00 commented:
Ok, the changes proposed, plus a modification for new games (win1252 by default, but the index is 2 now, and that must be changed). I would like to update the Font Editor on the fly in case the codepage is changed, but not sure if that is possible to do.

LaZar00 commented:
Great, thanks a lot. I had done a last update about the names to put in combolist for codepages, as we talked, but well, we can leave also as they are now.

I tried to do an update or something when changing codepage in combolist, and in case to have font editor, then, at least, "mark" the font tab to know that there has been a change. BUT I had no luck. I should research all the code to know how he does the "old" marked tab (the tab with diagonal line). So, I will leave the updates for now as they are. I've been using the 850 codepage and right now seems to work well. I would like to have some more feedback though from other users.

Kawa-oneechan commented:
Yeah, feedback would be nice but who'd look here? I'm gonna give this a few more days to simmer.

LaZar00 commented:
I added a minor update. Not sure if it is of your interest. I have the need to filter the files when opening/importing patches (or in batch). But I can not select specific files like only messages or only views. The code is there, but there was some type of typo or wrong coding. Who knows. I have updated it.
